src/vma/sock/sock-redirect.cpp

```diff
@@ -776,6 +819,9 @@ int setsockopt(int __fd, int __level, int __optname,
 		ret = p_socket_object->setsockopt(__level, __optname, __optval, __optlen);
 	}
 	else {
+		BULLSEYE_EXCLUDE_BLOCK_START
```

Comment:
i preferred calling just before it's needed
the reason i called in accept and connect at the the beginning of the function is that there is more then one flow calling the os function e.g. in connect
https://github.com/Mellanox/libvma/pull/311/files#diff-c49e7dd4d795b90a21b8c1708ff6d7e8L739
https://github.com/Mellanox/libvma/pull/311/files#diff-c49e7dd4d795b90a21b8c1708ff6d7e8L747
i wanted to keep it simple
the other option is to rewrite the connect logic to something like this (didn't test it and has an issue but that's the spirit)

```
if (__to && __to->sa_family == AF_INET && p_socket_object) {
	ret = p_socket_object->connect(__to, __tolen);
	if (p_socket_object->isPassthrough()) {
		handle_close(__fd, false, true);
		if (ret == 0) {
			srdr_logdbg_exit("failed (errno=%d %m)", errno);
			return ret;
		}

	}
}
if (p_socket_object) {
	p_socket_object->setPassthrough();
}
BULLSEYE_EXCLUDE_BLOCK_START
if (!orig_os_api.connect) get_orig_funcs();
BULLSEYE_EXCLUDE_BLOCK_END
ret = orig_os_api.connect(__fd, __to, __tolen);

if (ret >= 0)
	srdr_logdbg_exit("returned with %d", ret);
else
	srdr_logdbg_exit("failed (errno=%d %m)", errno);
return ret;
```